### PR TITLE
Update the operator and shipment of it

### PR DESCRIPTION
**Pull request by laxmikantbpandhare**
Reopens #117

### CODE REVIEW

**Comment by timflannagan** [comment]
(Checking if the recently added tide plugin is added to the list of checks)

File: content/en/docs/Tasks/update-and-ship-operator.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+title: "Update and ship an Operator"
+date: 2021-03-10
+weight: 2
+description: > 
+    In depth explanation of operator update and shipment process. 
+--- 
+
+## Introduction
+
+In the Operator Lifecycle Manager (OLM) ecosystem, the following resources are used to resolve Operator installations and upgrades:
```
Comment (camilamacedo86) [inline content/en/docs/Tasks/update-and-ship-operator.md:11]: See my suggestions in : https://github.com/operator-framework/olm-docs/pull/136/files

File: content/en/docs/Tasks/update-and-ship-operator.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
+---
+title: "Update and ship an Operator"
+date: 2021-03-10
+weight: 2
+description: > 
+    In depth explanation of operator update and shipment process.
```
Comment (camilamacedo86) [inline content/en/docs/Tasks/update-and-ship-operator.md:6]:
```suggestion
    This guide shows OLM users how to create upgrade paths for their operator bundles.
```

File: content/en/docs/Tasks/update-and-ship-operator.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+- Any z-stream release in Old CatalogSource will update to the latest z-stream release in New CatalogSource.
+
+- Unavailable releases can be considered "virtual" graph nodes; their content does not need to exist, the registry just needs to respond as if the graph looks like this.
+
+# Ship Operator
+
+For the shimpment of an operator, first make sure that manifests gets created with the updated version of operator. Check below link for mre details on operator manifest creation process.
+
+[How to create Operator manaifest?](/docs/tasks/creating-operator-manifests/)
```
Comment (camilamacedo86) [inline content/en/docs/Tasks/update-and-ship-operator.md:183]: The links should follow the standard. See that we should: 
- [text](alias)

And then, at the bottom of the page have the link alias such as; https://raw.githubusercontent.com/operator-framework/olm-docs/master/content/en/docs/best-practices/common.md

File: content/en/docs/Tasks/update-and-ship-operator.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+In the Operator Lifecycle Manager (OLM) ecosystem, the following resources are used to resolve Operator installations and upgrades:
+
+1. `ClusterServiceVersion (CSV)` - A YAML manifest created from Operator metadata that assists the Operator Lifecycle Manager (OLM) in running the Operator in a cluster.
+
+    A CSV is the metadata that accompanies an Operator container image, used to populate user interfaces with information like its logo, description, and version. It is also a source of technical information needed to run the Operator, like the RBAC rules it requires and which Custom Resources (CRs) it manages or depends on.
+
+    A CSV is composed of a Metadata, Install strategy, and CRDs.
+
+2. `CatalogSource` - Operator metadata, defined in CSVs, can be stored in a collection called a CatalogSource. CatalogSource contains metadata that OLM can query to discover and install operators with their dependencies.
```
Comment (camilamacedo86) [inline content/en/docs/Tasks/update-and-ship-operator.md:19]: I am not sure if these instructions are accurate. I think we need to explain how to the things with the Bundle Format.  See an example: https://raw.githubusercontent.com/operator-framework/olm-docs/master/content/en/docs/best-practices/common.md

File: content/en/docs/Tasks/update-and-ship-operator.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+In the above image, etcd is a package. Alpha and beta are the channels.
+
+Within a CatalogSource, Operators are organized into packages and streams of updates called channels, which should be a familiar update pattern from OpenShift Container Platform or other software on a continuous release cycle like web browsers.
+
+![Channels Image](/docs/Tasks/images/channels.png)
+
+`Figure 2. Packages and channels in a CatalogSource`
+
+A user indicates a particular package and channel in a particular CatalogSource in a Subscription.
```
Comment (camilamacedo86) [inline content/en/docs/Tasks/update-and-ship-operator.md:38]: The channel is defined in the index image see: https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/bundle/metadata/annotations.yaml#L7

Also, see here how to update the channels with sdk : https://master.sdk.operatorframework.io/docs/overview/cheat-sheet/#updating-bundle-channels

File: content/en/docs/Tasks/update-and-ship-operator.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
+---
+title: "Update and ship an Operator"
+date: 2021-03-10
+weight: 2
+description: > 
+    In depth explanation of operator update and shipment process. 
+---
```
Comment (camilamacedo86) [inline content/en/docs/Tasks/update-and-ship-operator.md:7]: IHMO we need to;
- explain how to do by as a how to' using the operator bundle format\
- we need to explain the manual steps but also we can describe how sdk cli tool can help the users. I would indeed first explain via sdk and then manually.

File: content/en/docs/Tasks/update-and-ship-operator.md (hunk 2 above)
Comment (camilamacedo86) [reply content/en/docs/Tasks/update-and-ship-operator.md:6]: WDYT?

File: content/en/docs/Tasks/update-and-ship-operator.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+title: "Update and ship an Operator"
+date: 2021-03-10
+weight: 2
+description: > 
+    In depth explanation of operator update and shipment process. 
+--- 
+
+## Introduction
```
Comment (camilamacedo86) [inline content/en/docs/Tasks/update-and-ship-operator.md:9]: I think Overview follows more the standard.

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
ihmo we need to shape more this one before get merged.

File: content/en/docs/Tasks/update-and-ship-operator.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+For this given scenario, OLM installs Operator version `0.1.2` to replace the existing Operator version `0.1.1`. Then, it installs Operator version `0.1.3` to replace the previously installed Operator version `0.1.2`. At this point, the installed operator version `0.1.3` matches the channel head and the upgrade is completed.
+
+![Graph Image](/docs/Tasks/images/graph.png)
+
+`Figure 3. OLM's graph of available channel updates`
+
+Each CSV has a replaces parameter that indicates which Operator it replaces. This builds a graph of CSVs that can be queried by OLM, and updates can be shared between channels. Channels can be thought of as entry points into the graph of updates.
+
+
+##### Channels in a package
```
Comment (camilamacedo86) [inline content/en/docs/Tasks/update-and-ship-operator.md:61]: IHMO we need to use the bundle format here.

File: content/en/docs/Tasks/update-and-ship-operator.md (hunk 5 above)
Comment (camilamacedo86) [reply content/en/docs/Tasks/update-and-ship-operator.md:38]: Also, the channels might feet better in https://github.com/operator-framework/olm-docs/pull/136 and here we could only describe the options to configure the paths and the upgrade paths.

**Comment by timflannagan** [comment]
Can we also squash into a single commit (or multiple if it warrants multiple commits) and update the PR title to be something more descriptive?

File: content/en/docs/Tasks/update-and-ship-operator.md (hunk 2 above)
Comment (laxmikantbpandhare) [reply content/en/docs/Tasks/update-and-ship-operator.md:6]: Sounds Perfect

File: content/en/docs/Tasks/update-and-ship-operator.md (hunk 3 above)
Comment (laxmikantbpandhare) [reply content/en/docs/Tasks/update-and-ship-operator.md:183]: Modified according to suggestion.

File: content/en/docs/Tasks/update-and-ship-operator.md (hunk 7 above)
Comment (laxmikantbpandhare) [reply content/en/docs/Tasks/update-and-ship-operator.md:9]: Agree. Updated as per suggestion.

File: content/en/docs/Tasks/update-and-ship-operator.md (hunk 2 above)
Comment (laxmikantbpandhare) [reply content/en/docs/Tasks/update-and-ship-operator.md:6]: Agree

**Pull request closed without merging**